src/apps/workspaces/crud/user_applet_access.py

```diff
@@ -671,20 +732,52 @@ async def get_workspace_managers(
             .where(
                 UserAppletAccessSchema.soft_exists(),
                 UserAppletAccessSchema.owner_id == owner_id,
-                UserAppletAccessSchema.role != Role.RESPONDENT,
                 has_access,
+                UserAppletAccessSchema.role != Role.RESPONDENT,
                 UserAppletAccessSchema.applet_id == applet_id if applet_id else True,
             )
-            .group_by(UserSchema.id, UserAppletAccessSchema.title)
+            .group_by(UserSchema.id)
+        ).cte("accepted_users")
+
+        query: Query = (
+            select(
+                func.coalesce(accepted_users.c.id, invited_users.c.id).label("id"),
+                func.coalesce(accepted_users.c.first_name, invited_users.c.first_name).label("first_name"),
+                func.coalesce(accepted_users.c.last_name, invited_users.c.last_name).label("last_name"),
+                func.coalesce(accepted_users.c.email_encrypted, invited_users.c.email).label("email_encrypted"),
+                func.coalesce(accepted_users.c.created_at, invited_users.c.created_at).label("created_at"),
+                func.coalesce(accepted_users.c.last_seen, invited_users.c.created_at).label("last_seen"),
+                func.coalesce(accepted_users.c.roles, invited_users.c.roles).label("roles"),
+                func.coalesce(accepted_users.c.applets, invited_users.c.applets, []).label("applets"),
+                func.coalesce(accepted_users.c.titles, invited_users.c.titles, []).label("titles"),
+                func.coalesce(invited_users.c.status, InvitationStatus.APPROVED).label("status"),
+                func.coalesce(accepted_users.c.is_pinned, False).label("is_pinned"),
+            )
+            .select_from(invited_users)
+            .join(accepted_users, accepted_users.c.id == invited_users.c.user_id, isouter=True, full=True)
         )
```

Comment:
Please ensure this case is processed correctly:
1. user is in the system, has role 'coordinator'
2. the same user is invited for 'reviewer' role. Status is 'pending'
What is just caught by my eyes is that  status in query result for such case would be 'pending', but the user is already in the system. It's because of 
`func.coalesce(invited_users.c.status, InvitationStatus.APPROVED).label("status")`
 
 Please check the other data validity taking into account this case.

Reply:
yeah, this is working well when is in applet context, but getting wrong status when are in workspace context, maybe the solution is only shows pending invitation when are in applet context

Reply:
Agreed; I don't think product/design has yet considered how to list Team members who have a pending invite in one applet, but are already approved in another applet. Might be worth asking product?

Reply:
@mbanting Any thoughts?

Reply:
Removed pending invitations on workspace context, keep it only in applet context.

Reply:
Thanks @rcmerlo, seems to be the most reasonable solution for now, as it's unclear how this could be sensibly handled otherwise from a product perspective (IMO). If requirements change around this, we can address it in a separate ticket.

Reply:
@rcmerlo
> this is working well when in applet context

A user can be invited to the applet with different roles. In my case, the user already has a `coordinator` role and has a pending invitation to the same applet for the `reviewer` role. As a result, the status shows as pending, which I believe is incorrect.

Reply:
@vshvechko I'm not sure how it's possible to invite the same user (= same email address) to the same applet multiple times (even if the roles are different). If I try to do so, the BE returns an error:

<img width="1453" alt="Screenshot 2024-06-13 at 9 51 44 AM" src="https://github.com/ChildMindInstitute/mindlogger-backend-refactor/assets/1282772/c20ad58a-ccf4-4827-8fc7-ed2eab5a0890">

How are you able to get around this restriction?

Reply:
@rcmerlo wow, looks like a bug. I'll ask QAs to check

Reply:
@rcmerlo you are right, sorry for frustration, new roles can be added via `edit access` functionality